asdf/asdf.py

```diff
@@ -112,6 +121,29 @@ def __init__(
             files follow custom conventions beyond those enforced by the
             standard.
         """
+        if ignore_version_mismatch is not None:
+            warnings.warn(
+                "The ignore_version_mismatch argument to AsdfFile will be removed in 3.0.",
+                AsdfDeprecationWarning,
+            )
+        else:
+            ignore_version_mismatch = True
+
+        if ignore_unrecognized_tag is not None:
+            warnings.warn(
+                "The ignore_unrecognized_tag argument to AsdfFile will be removed in 3.0. "
```

Comment:
Is there migration documentation this can link to? Also, I would say let's not commit to a version for removal here and instead just say it's deprecated.